### PR TITLE
Cherry-pick PRs 14394, 14424, 14500, 14740

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

Cherry-pick PRs #14394, #14424, #14500, #14740

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

- Clear element binding data from DYN upon Save As
- Don't save file to recent files list if Save As is canceled
- Add provision of tooltips to message boxes

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

**Comment by QilongTang** [comment]
Is the element binding one a must have? Because that cherry pick introduce new resource strings while the localization process is done for 2.19.5

File: src/DynamoCoreWpf/UI/Prompts/DynamoMessageBox.xaml [hunk 1]
```diff
@@ -5,6 +5,7 @@
         xmlns:ui="clr-namespace:Dynamo.UI"
         xmlns:localui="clr-namespace:Dynamo.Wpf.UI.GuidedTour"
         xmlns:w="clr-namespace:System.Windows;assembly=PresentationCore"
+        xmlns:fa5="http://schemas.fontawesome.com/icons/"
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/Prompts/DynamoMessageBox.xaml:8]: This line is causing build error I think @aparajit-pratap

**Comment by mjkkirschner** [comment]
Was the element binding change really made in 2.19? I thought that only went into 3.0?

**Comment by mjkkirschner** [comment]
> Is the element binding one a must have? Because that cherry pick introduce new resource strings while the localization process is done for 2.19.5

IMO - having the dialog which a user could translate themselves with more info is better than not having it - alternatively we could make it a known issue... which we also don't translate ;)

**Comment by aparajit-pratap** [comment]
> > Is the element binding one a must have? Because that cherry pick introduce new resource strings while the localization process is done for 2.19.5
> 
> IMO - having the dialog which a user could translate themselves with more info is better than not having it - alternatively we could make it a known issue... which we also don't translate ;)

I agree with having the messaging even if it's missing localization as it will be good to have something than nothing at all.

File: src/DynamoCoreWpf/UI/Prompts/DynamoMessageBox.xaml (hunk 1 above)
Comment (aparajit-pratap) [reply src/DynamoCoreWpf/UI/Prompts/DynamoMessageBox.xaml]: Thanks for pointing this out @QilongTang

**Pull request merged**